src/openforms/appointments/utils.py

```diff
@@ -82,18 +92,20 @@ def book_appointment_for_submission(submission: Submission) -> None:
         appointment_id = client.create_appointment(
             [product], location, start_at, appointment_client
         )
-        submission.appointment_information = f"Appointment Id: {appointment_id}"
         AppointmentInfo.objects.create(
             status=AppointmentDetailsStatus.success,
             appointment_id=appointment_id,
             submission=submission,
         )
-    except AppointmentCreateFailed as e:
+    except AppointmentCreateFailed:
         AppointmentInfo.objects.create(
             status=AppointmentDetailsStatus.failed,
             error_information="Failed to make appointment",
             submission=submission,
         )
+        raise AppointmentRegistrationFailed(
+            "Unable to create appointment", should_retry=True
+        )
```

Comment:
please always `raise NewException from original_exception` to keep all the context in the stack.